### DISCUSSION TITLE
Is it possible to run a task before running a neotest run?

### DISCUSSION THREAD

**Discussion by carlosflorencio** [stevearc/overseer.nvim#92] [Q&A] [answered]
Wondering if it's possible with the current plugin version to run a task (e.g builds the project) and only then run the neotest action?

Thanks!

**Comment by stevearc** [comment, marked as answer]
Should be! There's a component for specifying [dependencies for a task](https://github.com/stevearc/overseer.nvim/blob/master/doc/components.md#dependencies), so all you need to do is attach that to your neotest task. There's two ways to do this:
1. Add it to the `default_neotest` component alias
```lua
require("overseer").setup({
  component_aliases = {
    default_neotest = {
      { "dependencies", task_names = {
        { "shell", cmd = "sleep 4" },
      } },
      "default",
    },
  },
})
```
2. Pass it in to the neotest strategy spec
```lua
require("neotest").setup({
  strategies = {
    overseer = {
      components = {
        { "dependencies", task_names = {
          { "shell", cmd = "sleep 4" },
        } },
        "default_neotest",
      },
    },
  },
})
```
For this second option, you can also define the components as a function, so you can choose at runtime whether to add this dependency or not
```lua
require("neotest").setup({
  strategies = {
    overseer = {
      components = function(run_spec)
        return {
          { "dependencies", task_names = {
            { "shell", cmd = "sleep 4" },
          } },
          "default_neotest",
        }
      end,
    },
  },
})
```
Note that for these last two I pushed up a couple of changes just now, so you'll need to fetch the latest master.

**Reply by carlosflorencio** [reply]
Will go for the second option because those test dependencies are only required for certain projects (will choose at runtime), you're awesome! Thanks!
